pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1366,6 +1349,38 @@ func parseIpspec(ipspec *zconfig.Ipspec,
 		}
 		config.DnsServers = append(config.DnsServers, ds)
 	}
+	s := ipspec.GetSubnet()
+	if s == "" {
+		return nil
+	}
+	// Parse Subnet
+	_, subnet, err := net.ParseCIDR(s)
+	if err != nil {
+		errStr := fmt.Sprintf("bad subnet %s: %s", s, err)
+		return errors.New(errStr)
+	}
+	config.Subnet = *subnet
+	// default Dhcp/Gateway configuration
+	prefixLen, _ := config.Subnet.Mask.Size()
+	prefixRange := 0x01 << (32 - prefixLen)
+	start := 1
+	end := prefixRange - 3
+	if prefixRange >= types.LargeAppCount {
+		start = prefixRange / 2
+		end = prefixRange/2 - 3
+	}
+	// Parse Gateway
+	if g := ipspec.GetGateway(); g != "" {
+		config.Gateway = net.ParseIP(g)
+		if config.Gateway == nil {
+			errStr := fmt.Sprintf("bad gateway IP %s", g)
+			return errors.New(errStr)
+		}
+	} else if config.Subnet.IP != nil {
+		// set the default gateway
+		log.Warnf("No Gateway configuration, setting the default\n")
```

Comment:
Please include the value it is set to in the log message.

Reply:
done